fgpyo/util/metric.py

```diff
@@ -334,3 +365,104 @@ def fast_concat(*inputs: Path, output: Path) -> None:
             io.write_lines(
                 path=output, lines_to_write=list(io.read_lines(input_path))[1:], append=True
             )
+
+    @staticmethod
+    def _read_header(
+        reader: TextIOWrapper,
+        delimiter: str = "\t",
+        comment_prefix: str = "#",
+    ) -> MetricFileHeader:
+        """
+        Read the header from an open file.
+
+        The first row after any commented or empty lines will be used as the fieldnames.
+
+        Lines preceding the fieldnames will be returned in the `preamble`.
+
+        Args:
+            reader: An open, readable file handle.
+            file_format: A dataclass containing (at minimum) the file's delimiter and the string
+                prefixing any comment lines.
+
+        Returns:
+            A `MetricFileHeader` containing the field names and any preceding lines.
+
+        Raises:
+            ValueError: If the file was empty or contained only comments or empty lines.
+        """
+
+        preamble: List[str] = []
+
+        for line in reader:
+            if line.startswith(comment_prefix) or line.strip() == "":
+                preamble.append(line.strip())
+            else:
+                break
+        else:
+            raise ValueError(f"File {reader.name} did not contain a header line.")
+
+        fieldnames = line.strip().split(delimiter)
```

Comment:
probably want to check for an empty "line" since that's what we'd get if the file was empty (good thing to add a unit test for too)